node/txsimulator/txSimulator.go

```diff
@@ -0,0 +1,70 @@
+package txsimulator
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/node"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type transactionSimulator struct {
+	txProcessor     TransactionProcessor
+	accountsAdapter state.AccountsAdapter
+}
+
+// New returns a new instance of a transactionSimulator
+func New(txProcessor TransactionProcessor, accountsAdapter state.AccountsAdapter) (*transactionSimulator, error) {
```

Comment:
New what ?

Reply:
the package is called `txsimulator` sau the constructor's call will result in `txsimulator.New()`. `txsimulator.NewTxSimulator()` would have been redundant